src/picterra/client.py

```diff
@@ -155,25 +156,7 @@ def set_raster_detection_areas_from_file(self, raster_id, filename):
         )
         if not resp.ok:
             raise APIError(resp.text)
-        poll_interval = resp.json()['poll_interval']
-
-        # Wait for detection area to be associated with raster
-        def _is_ready():
-            logger.info('checking upload status')
-            resp = self.sess.get(
-                self._api_url('rasters/%s/detection_areas/upload/%s/' % (raster_id, upload_id))
-            )
-            if not resp.ok:
-                logger.warning('failed to get detection area status - retrying')
-                return False
-
-            if resp.json()['status'] == 'ready':
-                return True
-            elif resp.json()['status'] == 'failed':
-                raise APIError(resp.text)
-            else:
-                return False
-        _poll_at_interval(_is_ready, poll_interval)
+        self._wait_until_operation_completes(resp.json())
```

Comment:
Same question as for rasters, this is now returning an operation ?

Reply:
Yes, see answer above.